Sources/_Datadog_TestRunner/DDTestRunner.swift

```diff
@@ -0,0 +1,139 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+@_implementationOnly import XCTest
+
+internal class DDTestRunner: NSObject, XCTestObservation {
+    static var instance: DDTestRunner?
+
+    let testNameRegex = try? NSRegularExpression(pattern: "([\\w]+) ([\\w]+)", options: .caseInsensitive)
+    let supportsSkipping = NSClassFromString("XCTSkippedTestContext") != nil
+    var currentBundleName = ""
+    var activeTestSpan: DDSpan?
+
+    override init() {
+        super.init()
+        XCTestObservationCenter.shared.addTestObserver(self)
+        startTracer()
+    }
+
+    func startTracer() {
+        var clientToken = ""
+        if let envToken = ProcessInfo.processInfo.environment["DATADOG_CLIENT_TOKEN"] {
+            clientToken = envToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        } else if let bundleToken = Bundle.main.infoDictionary?["DatadogClientToken"] as? String {
+            clientToken = bundleToken.trimmingCharacters(in: CharacterSet.whitespacesAndNewlines)
+        }
+
+        if clientToken.isEmpty {
+            return
+        }
+
+        Datadog.initialize(
+            appContext: .init(),
+            configuration: Datadog.Configuration
+                .builderUsing(clientToken: clientToken, environment: CIEnvironmentValues.getEnvVariable("DD_ENV") ?? "test")
+                .set(serviceName: CIEnvironmentValues.getEnvVariable("DD_SERVICE") ?? ProcessInfo.processInfo.processName)
+                .set(tracedHosts: ["."])
+                .build()
+        )
+
+        Global.sharedTracer = Tracer.initialize(
```

Comment:
By initialising the `Global.sharedTracer` in tests instrumentation, we allow the target-under-test to overwrite it. I'm wondering if it's really needed for us to use the global tracer in tests instrumentation.

What is the "link" between the tests instrumentation and the target-under-test (TUD)? If it's only about starting the active span in DD Tests Runner (DDTR) and continuing it in TUD, maybe we only need to build a bridge between DDTR and TUD so that the first can inject the active span reference to the latter?

Reply:
The problem here, is that by definition in Open Tracing (and also OTel) there should only exist one tracer, we could avoid assigning the second tracer to shared if one already exists. Having two traces we would have also two different traces being sent and the injected headers would differ depending where the request is made. Also synchronizing two span generators for upload to backend could present some challenges.

Reply:
I see. I'm only worried if this is a scalable approach. Even now, the user may configure the state of his/her tracer (i.e. by customizing the `sendNetworkInfo` or passing `globalTags`):
```swift
Global.sharedTracer = Tracer.initialize(
    configuration: Tracer.Configuration(
        sendNetworkInfo: true,
        globalTags: ["foo": "bar"]
    )
)
```
If overwriting the `Global.sharedTracer` by our own, we will lost the user configuration. Do you see any nice solution for this problem?

> Having two traces we would have also two different traces being sent and the injected headers would differ depending where the request is made. Also synchronizing two span generators for upload to backend could present some challenges.

Instantiating a separate `Tracer`:
```swift
internal let testsTracer = Tracer(...)
```
doesn't instantiate separate serializer/uploader. The `storage` (serializer) and `upload` (uploader) components are managed by the `TracingFeature.instance`, which can be shared between multiple tracers (just like logging storage and upload can be shared among several loggers). The only challenge would be to pass the `testsTracer.activeSpan.context` value to the `Global.sharedTracer` when starting spans in the app under test.

Reply:
As I said in previous comment, we don't want to have the production tracer running on tests, and probably the user wont either, because receiving traces and logs from tests in their production endpoints probably will pollute their output. If they are already using datadog to trace their tests in an ad hoc setting, they are probably not interested in our solution or should need some transition in their code. 
About the configuration for the test tracer,  the idea for first version is all configuration being done by env variables, and evaluate in the future other ways

Reply:
All clear then 👌 - we'll handle doubled initialisation silently in unit tests.